cmd/skaffold/app/cmd/runner.go

```diff
@@ -72,17 +78,21 @@ func createNewRunner(out io.Writer, opts config.SkaffoldOptions) (runner.Runner,
 	return runner, configs, runCtx, nil
 }
 
-func runContext(out io.Writer, opts config.SkaffoldOptions) (*runcontext.RunContext, []*latestV1.SkaffoldConfig, error) {
+func runContext(out io.Writer, opts config.SkaffoldOptions) (*runcontext.RunContext, []util.VersionedConfig, error) {
 	configs, err := withFallbackConfig(out, opts, parser.GetAllConfigs)
 	if err != nil {
 		return nil, nil, err
 	}
-	setDefaultDeployer(configs)
 
-	if err := validation.Process(configs); err != nil {
+	// TODO: yuwen Check v1 or v2.
```

Comment:
should this be addressed in this PR?

Reply:
Maybe not, providing a v2 validation is not a small change and may deserve another PR.